pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -49,19 +57,79 @@ type PackageControllerClient struct {
 	// activeBundleTimeout is the timeout to activate a bundle on installation.
 	activeBundleTimeout time.Duration
 	valuesFileWriter    filewriter.FileWriter
+	// skipWaitForPackageBundle indicates whether the installer should wait
+	// until a package bundle is activated.
+	//
+	// Skipping the wait is desirable for full cluster lifecycle use cases,
+	// where resource creation and error reporting are asynchronous in nature.
+	skipWaitForPackageBundle bool
 }
 
 type ChartInstaller interface {
 	InstallChart(ctx context.Context, chart, ociURI, version, kubeconfigFilePath, namespace, valueFilePath string, values []string) error
 }
 
+// ChartInstallationDeleter handles deleting chart installations.
+type ChartInstallationDeleter interface {
+	Delete(ctx context.Context, kubeconfigFilePath, installName, namespace string) error
+}
+
+// NewPackageControllerClientFullLifecycle creates a PackageControllerClient
+// for the Full Cluster Lifecycle controller.
+//
+// It differs because the CLI use case has far more information available at
+// instantiation, while the FCL use case has less information at
+// instantiation, and the rest when cluster creation is triggered.
+func NewPackageControllerClientFullLifecycle(logger logr.Logger, chartInstaller ChartInstaller, deleter ChartInstallationDeleter, kubectl KubectlRunner) *PackageControllerClient {
+	return &PackageControllerClient{
+		chartInstaller:           chartInstaller,
+		deleter:                  deleter,
+		kubectl:                  kubectl,
+		skipWaitForPackageBundle: true,
+		eksaRegion:               eksaDefaultRegion,
+	}
+}
+
+// EnableFullLifecycle wraps Enable to handle run-time arguments.
+//
+// This method fills in the gaps between the original CLI use case, where all
+// information is known at PackageControllerClient initialization, and the
+// Full Cluster Lifecycle use case, where there's limited information at
+// initialization.
+func (pc *PackageControllerClient) EnableFullLifecycle(ctx context.Context, log logr.Logger, clusterName, kubeConfig string, chart *releasev1.Image, registryMirror *registrymirror.RegistryMirror, options ...PackageControllerClientOpt) (err error) {
+	log.V(6).Info("enabling curated package full lifecycle", "clusterName", clusterName)
+	defer func(err *error) {
+		if err != nil && *err != nil {
+			log.V(6).Error(*err, "enabling curated packages full lifecycle", "clusterName", clusterName)
+		} else {
+			log.V(6).Info("success enabling curated packages full lifecycle")
+		}
+	}(&err)
+	pc.skipWaitForPackageBundle = true
+	pc.clusterName = clusterName
+	pc.kubeConfig = kubeConfig
+	pc.chart = chart
+	pc.registryMirror = registryMirror
```

Comment:
This is not thread safe. If we decide to run our controller with more than one routine (I think right now it defaults to 1), this can become problematic. That's likely future, I wouldn't ask otherwise.

Is there any way we can avoid it?

Reply:
Hmm. I'll take a look. If it's as easy as wrapping things in a mutex, then it shouldn't be too bad to do.

Reply:
I have a solution in place. Doesn't seem too bad. Please let me know if you have ideas.